wallets/rn_cli_wallet_068_5/src/modals/PairModal.tsx

```diff
@@ -0,0 +1,199 @@
+import React from 'react';
+import {Text, View, StyleSheet} from 'react-native';
+import Modal from 'react-native-modal';
+import {AcceptRejectButton} from '../components/AcceptRejectButton';
+import {Events} from '../components/MethodsModal/Events';
+import {Methods} from '../components/MethodsModal/Methods';
+import {ModalHeader} from '../components/MethodsModal/ModalHeader';
+import {Tag} from '../components/Tag';
+
+interface PairModalProps {
+  proposal: any; //ToDo: fix.
+  visible: boolean;
+  open: (arg0: boolean) => void;
+  handleAccept: () => void;
+}
+
+/*
+     @notice: Proposal Modal for initiating the pair()
+     @params: proposal, visible, open, handleAccept
+
+     Rendering
+      1. ModalHeader
+      2. Requested Permissions Text
+      3. Chain + Methods + Events
+      4. Accept/Reject Buttons
+  */
+
+export function PairModal({
+  proposal,
+  visible,
+  open,
+  handleAccept,
+}: PairModalProps) {
+  const name = proposal?.params?.proposer?.metadata?.name;
+  const url = proposal?.params?.proposer?.metadata.url;
+  const methods = proposal?.params?.requiredNamespaces.eip155.methods;
```

Comment:
Is the modal always expecting `eip155` namespace only?

Reply:
Going to update on next sprint for other nameSpaces